test/unit/range/iterator_test_template.hpp

```diff
@@ -642,6 +651,69 @@ TYPED_TEST_P(iterator_fixture, compare_geq)
     }
 }
 
+// ---------------------------------------------------------------------------------------------------------------------
+// Contiguous Iterator
+// ---------------------------------------------------------------------------------------------------------------------
+
+TYPED_TEST_P(iterator_fixture, address_test)
+{
+    if constexpr (std::derived_from<typename TestFixture::iterator_tag, std::contiguous_iterator_tag>)
+    {
+        auto begin_it = std::ranges::begin(this->test_range);
+        EXPECT_EQ(std::to_address(begin_it), std::addressof(*begin_it));
+    }
+
+    if constexpr (std::derived_from<typename TestFixture::iterator_tag, std::contiguous_iterator_tag> &&
+                  TestFixture::const_iterable)
+    {
+        auto cbegin_it = std::ranges::cbegin(this->test_range);
+        EXPECT_EQ(std::to_address(cbegin_it), std::addressof(*cbegin_it));
+    }
+}
+
+template <typename it_begin_t, typename it_sentinel_t>
+inline void address_difference_test(it_begin_t && it_begin, it_sentinel_t && it_end)
```

Comment:
This is more a general question: Do we add `inline` for template functions that are anyways `inline`?

Reply:
I'd say it is not always trivial. But in general we follow this here: https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Rs-inline

Even though funciton templates are usually inline: https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Rf-inline, we put it in front of free functions and exertnally defined member functions, since we don't worry about ABI. This relieves the programmer to not think about why or why not a function was declared inline.